packages/@tailwindcss-upgrade/src/codemods/template/migrate-arbitrary-utilities.ts

```diff
@@ -0,0 +1,340 @@
+              yield replacementCandidate
+            }
+          }
+
+          // Try bare value based on the `--spacing` value. E.g.:
+          //
+          // - `w-[64rem]` → `w-256`
+          if (spacingMultiplier !== null) {
+            for (let replacementCandidate of parseCandidate(
+              designSystem,
+              `${root}-${spacingMultiplier}`,
+            )) {
+              yield replacementCandidate
+            }
+
+            // Try bare value based on the `--spacing` value, but with a modifier
+            if (candidate.modifier) {
+              for (let replacementCandidate of parseCandidate(
+                designSystem,
+                `${root}-${spacingMultiplier}${candidate.modifier}`,
+              )) {
+                yield replacementCandidate
+              }
+            }
+          }
+
+          // Try as arbitrary value
+          for (let replacementCandidate of parseCandidate(designSystem, `${root}-[${value}]`)) {
+            yield replacementCandidate
+          }
+
+          // Try as arbitrary value with modifier
+          if (candidate.modifier) {
+            for (let replacementCandidate of parseCandidate(
+              designSystem,
+              `${root}-[${value}]${printModifier(candidate.modifier)}`,
+            )) {
+              yield replacementCandidate
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+function parseCandidate(designSystem: DesignSystem, input: string) {
+  return designSystem.parseCandidate(
+    designSystem.theme.prefix && !input.startsWith(`${designSystem.theme.prefix}:`)
+      ? `${designSystem.theme.prefix}:${input}`
+      : input,
+  )
+}
+
+// Let's make sure that all variables used in the value are also all used in the
+// found replacement. If not, then we are dealing with a different namespace or
+// we could lose functionality in case the variable was changed higher up in the
+// DOM tree.
+function allVariablesAreUsed(
+  designSystem: DesignSystem,
+  candidate: Candidate,
+  replacement: Candidate,
+) {
+  let value: string | null = null
+
+  // Functional utility with arbitrary value and variables
+  if (
+    candidate.kind === 'functional' &&
+    candidate.value?.kind === 'arbitrary' &&
+    candidate.value.value.includes('var(--')
+  ) {
+    value = candidate.value.value
+  }
+
+  // Arbitrary property with variables
+  else if (candidate.kind === 'arbitrary' && candidate.value.includes('var(--')) {
+    value = candidate.value
+  }
+
+  // No variables in the value, so this is a safe migration
+  if (value === null) {
+    return true
+  }
+
+  let replacementAsCss = designSystem
+    .candidatesToCss([designSystem.printCandidate(replacement)])
+    .join('\n')
+
+  let isSafeMigration = true
+  ValueParser.walk(ValueParser.parse(value), (node) => {
+    if (node.kind === 'function' && node.value === 'var') {
+      let variable = node.nodes[0].value
+      let r = new RegExp(`var\\(${variable}[,)]\\s*`, 'g')
```

Comment:
Does this need to handle eventual space around the variable name? e.g.

```css
.foo {
  color: var( --red);
  color: var(--red );
}
```

etc.

Reply:
This shouldn't be necessary because the values are already canonicalized to their minimal form. So the whitespace inconsistencies should be handled already.